doc/design/model_market.md

```diff
@@ -0,0 +1,119 @@
+# Model Market
+
+The model market is a web site where model developers can publish and share their models and analysts can find some useful models to finish the analysis work. The model market can be deployed anywhere like on the cloud or on-premise with some configurations.
+
+In the [model zoo design](model_zoo.md), we described how model developers develop, publish and share custom models on SQLFlow, and how analysts can make use of the shared model with SQLFlow.
+
+Here, we'll describe how to build the model market model developers and analysts.
+
+## Overview
+
+The model market is designed to:
+
+1. Manage user login and logout.
+1. View published images available for the current user.
+1. View published trained models available for the current user.
+1. Publish (or remove) model definition images.
+1. Publish (or remove) trained models.
+1. Share (or remove share) model definition images to other users.
+1. Share (or remove share) trained models to other users.
+
+<p align="center">
+<img src="figures/model_market_overview.png">
+</p>
+
+In order to support publishing and sharing models securely, the model market is able to communicate with some [SSO](https://en.wikipedia.org/wiki/Single_sign-on) service to authenticate users. Only users that are logged in can do operations on the model market.
+
+Then model definition images and trained model can have below accessibility settings:
```

Comment:
I am afraid that model definitions and trained models need to have different access control lists.  A model definition from a published paper is likely public, but a model trained with private data should be private.

Reply:
Yes, in this part I was trying to explain the types of visibility settings. Model definitions and trained models' accessibility settings indeed need to have different access control lists. I'll make it clear.